standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -913,74 +900,6 @@ private static void logAndAudit(final String m) {
     logAuditEvent(m);
   }
 
-  private String startFunction(String function, String extraLogInfo) {
-    incrementCounter(function);
-    logAndAudit((getThreadLocalIpAddress() == null ? "" : "source:" + getThreadLocalIpAddress() + " ") +
-        function + extraLogInfo);
-    com.codahale.metrics.Timer timer =
-        Metrics.getOrCreateTimer(MetricsConstants.API_PREFIX + function);
```

Comment:
I wonder if its an error if we already have a timer running for a method already - I think in normal cases we may have only one api call running in a single connection...

Reply:
yes, we have double counted the metrics of api calling by the startFunction and the PerfLogger, e.g, the total call count and the active count of the api.